BlueprintUI/Sources/Blueprint View/BlueprintView.swift

```diff
@@ -27,44 +27,84 @@ import UIKit
 /// ```
 public final class BlueprintView: UIView {
 
-    private var needsViewHierarchyUpdate: Bool = true
-    private var hasUpdatedViewHierarchy: Bool = false
-    private var lastViewHierarchyUpdateBounds: CGRect = .zero
+    private var neededUpdateType : UpdateType
+    
+    private var lastViewHierarchyUpdateBounds: CGRect
 
     /// Used to detect reentrant updates
     private var isInsideUpdate: Bool = false
 
     private let rootController: NativeViewController
-
+    
     /// The root element that is displayed within the view.
     public var element: Element? {
-        didSet {
-            setNeedsViewHierarchyUpdate()
+        set {
+            self.set(element: newValue)
+        }
+        get {
+            return self._element
+        }
+    }
+    
+    private var _element : Element?
+    
+    public func set(animated : Bool = UIView.isInAnimationBlock, element: Element?)
```

Comment:
Correct – since this is the "seam" between declarative and non-declarative land – you may be using a blueprint view within your normal UIKit views, which is what this is. Maybe I should remove the param and just rely on if we're in an animation block?

Reply:
Going to add an `Animated` element as well, but also leave this – both serve their purpose depending on the layer of interaction (UIView layer vs. within the element tree).